### ISSUE TITLE
[BUG] `avoid-dynamic` doesn't warn about implicit `dynamic` in function parameters

### ISSUE THREAD

**Issue by DetachHead** [dart-code-checker/dart-code-metrics#750]
* **Dart code metrics version:** 4.13.0
* **Dart sdk version:** 2.16.1

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
analyzer:
  plugins:
    - dart_code_metrics

dart_code_metrics:
  rules:
    - avoid-dynamic
```

</details>

**What did you do? Please include the source code example causing the issue.**

```dart
void foo(a) {
  print(a);
}
```

**What did you expect to happen?**
warning on `a` parameter
**What actually happened?**
no warning
```
>flutter pub run dart_code_metrics:metrics analyze lib
✔ no issues found!
```
![image](https://user-images.githubusercontent.com/57028336/159166433-7c177830-2344-4d3b-a7ed-c9a353a50b52.png)

**Comment by KotlinIsland** [comment]
`dynamic`: No way! That's nonsafe (safety-denying) software.

**Comment by incendial** [comment]
You should have a build-in check for dynamics:
<img width="679" alt="Screenshot 2022-03-22 at 21 18 06" src="https://user-images.githubusercontent.com/21116742/159548923-02ef7cf0-85c9-49d8-a4a6-d8b54265b013.png">

Can be set up with `implicit-dynamic`:
<img width="441" alt="Screenshot 2022-03-22 at 21 19 27" src="https://user-images.githubusercontent.com/21116742/159549144-17c02940-c83c-43a7-9994-1d66663d1412.png">
